src/main/java/com/redhat/devtools/lsp4ij/server/definition/launching/UserDefinedLanguageListener.java

```diff
@@ -53,18 +53,33 @@ public void handleStatusChanged(LanguageServerWrapper languageServer) {
 
     @Override
     public void handleChanged(@NotNull LanguageServerChangedEvent event) {
-        if (event.configurationChanged && event.serverDefinition == serverDefinition) {
-            // Case 2: configuration has changed:
-            // Try to get the user defined configuration and
-            // push it with 'workspaceService/didChangeConfiguration' to the all started language servers
-            // which matches the server definition.
-            LanguageServiceAccessor.getInstance(project)
-                    .getStartedServers()
-                    .forEach(ls -> {
-                        if (ls.serverDefinition == serverDefinition) {
-                            didChangeConfiguration(ls);
-                        }
-                    });
+        if (event.serverDefinition == serverDefinition){
+            if (event.configurationChanged) {
+                // Case 2: configuration has changed:
+                // Try to get the user defined configuration and
+                // push it with 'workspaceService/didChangeConfiguration' to the all started language servers
+                // which matches the server definition.
+                LanguageServiceAccessor.getInstance(project)
+                        .getStartedServers()
+                        .forEach(ls -> {
+                            if (ls.serverDefinition == serverDefinition) {
+                                didChangeConfiguration(ls);
+                            }
+                        });
+            }
+            if (event.initializationOptionsContentChanged) {
+                // Case 2: initializationOption has changed:
+                // Try to get the user defined initializationOption and
+                // restart all started language servers
+                // which matches the server definition.
+                LanguageServiceAccessor.getInstance(project)
+                        .getStartedServers()
+                        .forEach(ls -> {
+                            if (ls.serverDefinition == serverDefinition) {
+                                ls.restart();
+                            }
+                        });
+            }

```

Comment:
should be put 1st. If init options changed, servers are restarted, then triggering didChangeConfiguration is unnecessary